File: backend/app/schemas/expense.py
import uuid
from datetime import datetime
from decimal import Decimal

from sqlmodel import SQLModel

from app.enums import ExpenseCategory
from app.models import ExpenseBase


class ExpensePublic(ExpenseBase):
    id: uuid.UUID
    account_name: str
    
class ExpenseCreate(SQLModel):
    date: datetime
    category: ExpenseCategory
    amount: Decimal
    description: str
    fund_source: uuid.UUID
    
class ExpenseSummary(SQLModel):
    total: Decimal
    count: int
    largest: ExpensePublic | None
    
    
class ExpenseByCategory(SQLModel):
    category: ExpenseCategory
    amount: Decimal
    
    
class ExpenseList(SQLModel):
    items: list[ExpensePublic]
    total_items: int
    summary: ExpenseSummary
    expense_by_category: list[ExpenseByCategory]

File: frontend/src/types/expense.ts
export type ExpensePeriod =
    | 'today'
    | 'this_week'
    | 'this_month'
    | 'last_month'
    | 'this_year'
    | 'all'

export type ExpenseCategory = 
    | 'Food'
    | 'Maintenance'
    | 'Utilities'
    | 'Transportation'
    | 'Supplies'
    | 'Payroll'
    | 'Benefits'
    | 'Production'
    | 'Miscellaneous'
    | 'Equipment'

interface ExpenseBase {
    date: string
    category: string
    description: string
}

export interface Expense extends ExpenseBase {
    id: string
    amount: string
    account_name: string
}

export interface ExpenseSummary {
    total: string
    count: number
    largest: Expense | null
}

export interface ExpenseByCategory {
    category: string
    amount: string
}

export interface ExpenseList {
    items: Expense[]
    total_items: number
    summary: ExpenseSummary
    expense_by_category: ExpenseByCategory[]
}